minimint/src/net/api.rs

```diff
@@ -67,44 +84,57 @@ where
                         (*param, value)
                     })
                     .collect();
-                let module: &M = req.state().minimint.as_ref().into();
+                let module: &M = req.state().minimint.as_ref().as_ref();
                 (endpoint.handler)(module, params, data)
             });
     }
 }
 
-#[instrument(skip_all)]
-async fn submit_transaction(mut req: Request<State>) -> tide::Result {
-    trace!(?req, "Received API request");
-    let transaction: Transaction = req.body_json().await?;
-    let tx_id = transaction.tx_hash();
-    debug!("Sending peg-in request to consensus");
-    req.state()
-        .minimint
-        .submit_transaction(transaction)
-        .expect("Could not submit sign request to consensus");
+fn server_endpoints() -> &'static [ApiEndpoint<MinimintConsensus<rand::rngs::OsRng>>] {
+    &[
+        ApiEndpoint {
+            path_spec: "/transaction",
+            params: &[],
+            method: http::Method::Put,
+            handler: |minimint, _params, transaction| {
+                let transaction: Transaction =
+                    match serde_json::from_slice(&serde_json::to_vec(&transaction).unwrap()) {
+                        Ok(t) => t,
+                        Err(_) => return Ok(http::Response::new(400)),
+                    };
+                let tx_id = transaction.tx_hash();
 
-    // TODO: give feedback in case of error
-    let body = Body::from_json(&tx_id).expect("encoding error");
```

Comment:
Do errors in the request parsing kill the API thread?

Reply:
no, this errors in serialization of response into json, which should not happen